sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeCreator.scala

```diff
@@ -312,6 +312,10 @@ case object NamePlaceholder extends LeafExpression with Unevaluable {
  * Returns a Row containing the evaluation of all children expressions.
  */
 object CreateStruct extends FunctionBuilder {
+  /**
+   * Returns a named struct with generating names or using the names when available.
+   * It should be used only for an internal purpose.
```

Comment:
Everything in catalyst is private right?

Reply:
Hm, I will rephrase it to clarify like it shouldn't be used when `struct` is explicitly specified by a user.

Reply:
You could also consider moving this into the `CreateNamedStruct` companion to avoid confusion. BTW in which places do we call this method and still want to retain the old name?

Reply:
There are some cases when `CreateNamedStruct` is inserted, e.g.) https://github.com/apache/spark/pull/28633#discussion_r429756482

I was thinking that it's fine to treat the cases as just using `named_struct` internally. Also, it makes many diff in SQL tests and etc. Thought it's better to minimise the diff.

Actually moving to `CreateNamedStruct` is the first way I tried. One thing is that `CreateNamedStruct` case class has the same signature so `CreateNamedStruct` companion object can't have the same signature at `apply`.

I could consistently have `CreateNamedStruct.create` and `CreateStruct.create` if you prefer this way.

Reply:
I pushed some changes but let me know if you prefer this way. I don't mind changing it.